pkg/cli/debug_job_trace.go

```diff
@@ -73,7 +81,91 @@ func getJobTraceID(sqlConn *sqlConn, jobID int64) (int64, error) {
 	return traceID, nil
 }
 
-func writeTracePayloadJSON(sqlConn *sqlConn, jobID int64, traceFilePath string) error {
+type inflightTraceRow struct {
+	nodeID     int64
+	rootOpName string
+	traceJSON  string
+	traceStr   string
+	jaegerJSON string
+}
+
+// stitchJaegerJSON adds the trace spans from jaegerJSON into the
+// nodeTraceCollection object, and returns a new cumulative
+// tracing.TraceCollection object.
+func stitchJaegerJSON(
+	nodeTraceCollection *tracing.TraceCollection, jaegerJSON string,
+) (*tracing.TraceCollection, error) {
+	var cumulativeTraceCollection *tracing.TraceCollection
+
+	// Unmarshal the jaegerJSON string to a TraceCollection.
+	var curTraceCollection tracing.TraceCollection
+	if err := json.Unmarshal([]byte(jaegerJSON), &curTraceCollection); err != nil {
+		return cumulativeTraceCollection, err
+	}
+
+	// Sanity check that the TraceCollection has a single trace entry.
+	if len(curTraceCollection.Data) != 1 {
+		return cumulativeTraceCollection, errors.AssertionFailedf("expected a single trace but found %d",
+			len(curTraceCollection.Data))
+	}
+
+	if nodeTraceCollection == nil {
```

Comment:
Can we move this case further up the method to be the first check?

Reply:
Hmmm we do need to unmarshal the curTraceCollection before we can assign cumulativeTraceCollection to it. In other words, we're not doing any extra work between here and the beginning of the method. Unless your suggestion was a code restructure?

Reply:
Ah, nevermind. Just got myself confused.